src/utils/ApiClient.js

```diff
@@ -347,4 +349,13 @@ export default class ApiClient {
             });
         };
     }
+
+    /**
+     * Helper function to check if the response body is an empty object
+     *
+     * @private
+     */
+    isNotEmptyBody(body) {
+        return !(Object.keys(body).length === 0 && Object.getPrototypeOf(body) === Object.prototype);
+    }
```

Comment:
I suggest to complete the util like 
create a function `isEmptyResponseBody` and then create another function `isNotEmptyResponseBody` that calls the latter